### PR TITLE
cmd: don't bind debug endpoints by default

### PR DESCRIPTION
**Pull request by gsora**
Separate Prometheus from pprof/QBFT HTTP endpoint, and let user configure the listening address for them.

Added the `--debug-address` CLI argument, so that user can specify the IP/port combo that suite them better.

If no `--debug-address` is specified, don't bind debug endpoints.

category: refactor
ticket: none

### CODE REVIEW

**Comment by OisinKyne** [comment]
Needs a corresponding PR here: https://github.com/ObolNetwork/obol-docs/blob/main/docs/charon/charon-cli-reference.md#the-run-subcommand

**Review by dB2510 — APPROVED** [review]
We will need to update our core team cluster and [CDVN](https://github.com/ObolNetwork/charon-distributed-validator-node/blob/3a343b6ca1199f9fcee954e69ee39bb786556e0c/compose-debug.yml#L48) in line with this change.

File: docs/configuration.md [hunk 1]
```diff
@@ -139,6 +139,7 @@ Usage:
 Flags:
       --beacon-node-endpoints strings      Comma separated list of one or more beacon node endpoint URLs.
       --builder-api                        Enables the builder api. Will only produce builder blocks. Builder API must also be enabled on the validator client. Beacon node must be connected to a builder-relay to access the builder network.
+      --debug-address string               Listening address (ip and port) for the pprof and QBFT debug API.
```
Comment (pinebit) [inline docs/configuration.md:142]: Worth to mention it is disabled by default.

**Comment by gsora** [comment]
https://github.com/ObolNetwork/obol-docs/pull/299 docs PR

**Review by xenowits — APPROVED** [review]
LGTM!

**Pull request merged**